components-core/src/main/java/com/adyen/checkout/components/core/internal/analytics/data/DefaultNewAnalyticsRepository.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2024 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ararat on 28/2/2024.
+ */
+
+package com.adyen.checkout.components.core.internal.analytics.data
+
+import com.adyen.checkout.components.core.internal.analytics.AnalyticsEvent
+import com.adyen.checkout.components.core.internal.analytics.data.local.AnalyticsLocalDataStore
+import com.adyen.checkout.components.core.internal.analytics.data.remote.AnalyticsRemoteDataStore
+import com.adyen.checkout.components.core.internal.analytics.data.remote.AnalyticsSetupProvider
+import com.adyen.checkout.components.core.internal.analytics.data.remote.AnalyticsTrackRequestProvider
+
+internal class DefaultNewAnalyticsRepository(
+    private val localInfoDataStore: AnalyticsLocalDataStore<AnalyticsEvent.Info>,
```

Comment:
We discussed this with @OscarSpruit and it would not add much value for us, since the repository implementation is quite simple already. Also we can not really remove the dependency on analytics types if we add a new abstraction.
I would only do that if the logic gets more complicated or the analytics behavior changes. Wdyt?